contracts/non-fungible-tokens/andromeda-crowdfund/src/testing/tests.rs

```diff
@@ -111,1910 +41,2280 @@ fn init(deps: DepsMut, modules: Option<Vec<Module>>) -> Response {
 fn test_instantiate() {
     let mut deps = mock_dependencies_custom(&[]);
 
-    let modules = vec![Module {
-        name: Some(RATES.to_owned()),
-        address: AndrAddr::from_string(MOCK_RATES_CONTRACT.to_owned()),
-        is_mutable: false,
-    }];
-
-    let res = init(deps.as_mut(), Some(modules));
+    let res = init(deps.as_mut(), None);
 
     assert_eq!(
         Response::new()
             .add_attribute("method", "instantiate")
             .add_attribute("type", "crowdfund")
             .add_attribute("kernel_address", MOCK_KERNEL_CONTRACT)
-            .add_attribute("owner", "owner")
-            .add_attribute("action", "register_module")
-            .add_attribute("module_idx", "1"),
+            .add_attribute("owner", "owner"),
         res
     );
 
     assert_eq!(
-        Config {
-            token_address: AndrAddr::from_string(MOCK_TOKEN_CONTRACT.to_owned()),
-            can_mint_after_sale: true
-        },
-        CONFIG.load(deps.as_mut().storage).unwrap()
+        mock_campaign_config(),
+        CAMPAIGN_CONFIG.load(deps.as_mut().storage).unwrap()
     );
-
-    assert!(!SALE_CONDUCTED.load(deps.as_mut().storage).unwrap());
-}
-
-#[test]
-fn test_mint_unauthorized() {
-    let mut deps = mock_dependencies_custom(&[]);
-    init(deps.as_mut(), None);
-
-    let msg = ExecuteMsg::Mint(vec![CrowdfundMintMsg {
-        token_id: "token_id".to_string(),
-        owner: None,
-        token_uri: None,
-        extension: TokenExtension {
-            publisher: "publisher".to_string(),
-        },
-    }]);
-    let info = mock_info("not_owner", &[]);
-    let res = execute(deps.as_mut(), mock_env(), info, msg);
-
-    assert_eq!(ContractError::Unauthorized {}, res.unwrap_err());
 }
 
 #[test]
-fn test_mint_owner_not_crowdfund() {
+fn test_instantiate_invalid_tiers() {
```

Comment:
https://github.com/andromedaprotocol/andromeda-core/blob/development/packages/std/src/os/vfs.rs#L273

In general it might be simpler to structure tests in thise case-like structure. That way we can add/edit test cases very easily.

Reply:
Refactored tests